src/services/user/registration.js

```diff
@@ -32,26 +32,27 @@ const populateUser = (app, data) => {
 
 	if (data.classId) user.classId = data.classId;
 
-	if (data.importHash) {
-		return app.service('users').find({ query: { importHash: data.importHash, _id: data.userId, $populate: ['roles'] } }).then((users) => {
-			if (users.data.length <= 0 || users.data.length > 1) {
-				throw new errors.BadRequest('Kein Nutzer für die eingegebenen Daten gefunden.');
-			}
-			oldUser = users.data[0];
-
-			Object.keys(oldUser).forEach((key) => {
-				if (oldUser[key] !== null && key !== 'firstName' && key !== 'lastName') {
-					user[key] = oldUser[key];
-				}
-			});
+	if (!data.importHash) {
+		return Promise.reject('Ungültiger Link');
+	}
 
-			user.roles = user.roles.map((role) => (typeof role === 'object' ? role.name : role));
+	return app.service('users').find({ query: { importHash: data.importHash.toString(), _id: data.userId.toString(), $populate: ['roles'] } }).then((users) => {
```

Comment:
Codacy found an issue: [This line has a length of 160. Maximum allowed is 120.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5749742)